runtime/connectors/connectors.go

```diff
@@ -79,7 +79,19 @@ func (ps PropertySchema) ValidateType(val any) bool {
 type Env struct {
 	RepoDriver string
 	RepoDSN    string
-	Variables  map[string]string
+	// user provided env variables
+	// following env variables are used in connectors :
+	// use_host_credentials - disable/enable use of host credentials :
+	//
+	// gcs specific variables :
+	// gcs_credentials_file : file where credentials are stored in json format
+	// gcs_credentials : credential file content in json format
+	//
+	// s3 specific variables :
+	// aws_access_key_id
+	// aws_secret_access_key
+	// aws_session_token
```

Comment:
Let's document these in a docstring in the `gcs` and `s3` packages separately (maybe on the `ConsumeAsIterator` function?). Then we'll remember to keep them updated when making changes in the future (and each connector package becomes the source of truth for what options it has).

Reply:
The idea was to keep them at a central place since I expect them to be put into some external documentation at some point.
I can also add some docstring on `ConsumeAsIterator` focussing on credentials where we can document the env variables.